### PR TITLE
feat: account heads changed along with journal entry type and descrip…

### PR DESCRIPTION
**Pull request by rahulgupta8848**
## Background/Context

- On the submission of asset value adjustment, automatically journal entry is passed in which depreciation and accumulated depreciation accounts are used to adjust the value

- Also, the negative amount is passed in the journal entry in case of increase in asset value.
- Journal entry type is of depreciation instead of journal entry.
- Remark of journal entry is of depreciation entry instead of revaluaton entry.
- Accounting Dimension values not getting updated

## Solution Summary
- Provide a field named “Difference account” and one can choose the revaluation or impairment account as may be the case differ
- No negative entry is allowed in the journal entry passed.
- Journal entry type is to be set to journal entry.
- Remark of journal entry is to be set as revaluation entry.
- Accounting Dimension values getting fetched from asset and updated in Journal Entry

**1. In the Asset Value Adjustment Doctype, we have added a new ‘Difference Account’ box.**
image (12) (1)

**2. When the value of asset is increasing, the fixed asset account from the asset category is debited and the difference account mentioned at the asset value adjustment doctype is credited. The opposite will happen when  the asset value is decreasing. Also, the entry type is changed from Depriciation Entry to Journal Entry**
image (13) (1)

**3. Remark is changed from Depriciation Entry to Revaluation Entry**
image (14) (1)

### CODE REVIEW

File: erpnext/assets/doctype/asset_value_adjustment/asset_value_adjustment.py
Comment (nabinhait) [inline erpnext/assets/doctype/asset_value_adjustment/asset_value_adjustment.py:84]: Set `value_after_depreciation` on submission, instead of validate function.

File: erpnext/assets/doctype/asset_value_adjustment/asset_value_adjustment.py
Comment (nabinhait) [inline erpnext/assets/doctype/asset_value_adjustment/asset_value_adjustment.py]: This is not required as dimension values are already set in lines 138-153.

File: erpnext/assets/doctype/asset_value_adjustment/asset_value_adjustment.py
Comment (nabinhait) [inline erpnext/assets/doctype/asset_value_adjustment/asset_value_adjustment.py]: Rename the function to `get_value_of_accounting_dimensions`

File: erpnext/assets/doctype/asset_value_adjustment/asset_value_adjustment.py
Comment (nabinhait) [inline erpnext/assets/doctype/asset_value_adjustment/asset_value_adjustment.py]: Avoid `get_value` inside a for loop, here you can fetch all the dimensions in a single query.

You can write this function like below:

```
dimension_fields = frappe.get_list("Accounting Dimension", pluck="fieldname") + ['cost_center']
return frappe.db.get_value("Asset", asset_name, fieldname=dimension_fields, as_dict=True)
```

File: erpnext/assets/doctype/asset_value_adjustment/test_asset_value_adjustment.py
Comment (nabinhait) [inline erpnext/assets/doctype/asset_value_adjustment/test_asset_value_adjustment.py:294]: Also, check the value of `value_after_depreciation` in Asset

File: erpnext/assets/doctype/asset_value_adjustment/asset_value_adjustment.py
Comment (nabinhait) [inline erpnext/assets/doctype/asset_value_adjustment/asset_value_adjustment.py:81]: As we are reversing the sign of difference_amount value (which is correct), we need to write a patch to update the value of existing asset value adjustment records.

You can refer https://github.com/frappe/erpnext/pull/40390/files#diff-c3d346867df08838dc46256708b1cce71ed729e35ff7032e4eef16b752242a80

**Review by nabinhait — CHANGES_REQUESTED** [review]

**Comment by nabinhait** [comment]
Please update the documentation of the asset value adjustment (https://docs.erpnext.com/docs/user/manual/en/asset-value-adjustment). Currently, it's very basic, let's explain which account will be affected in case of an increase/decrease in asset value.

**Comment by rahulgupta8848** [comment]
@nabinhait, I have made all the changes you have asked for.
Should I update the document now, even if the feature is not built yet, or wait until after the merge?

**Review by nabinhait — APPROVED** [review]

**Comment by nabinhait** [comment]
> @nabinhait, I have made all the changes you have asked for. Should I update the document now, even if the feature is not built yet, or wait until after the merge?

Please make the changes in the docs

**Pull request merged**
